src/zarr/storage/_obstore.py

```diff
@@ -145,15 +147,15 @@ async def set(self, key: str, value: Buffer) -> None:
 
         self._check_writable()
 
-        buf = value.to_bytes()
+        buf = value.as_numpy_array().view(np.uint8)
```

Comment:
Is it worthwhile to make this a standard method? This would make it easier to use throughout the codebase

Some possible names:

* `to_bytelike`
* `to_binary`
* `to_uint8s`
* ?

Reply:
i guess i'm also curious to know why we have a `to_bytes()` method that makes a copy, when `.as_numpy_array().view(np.uint8)` achieves the same effective result (an iterable of bytes) without a copy.

Reply:
This was suggested in #2925, so I've added an `as_bytes_like` method.

Reply:
`bytes` objects always own their memory. So creating a new `bytes` object means making a copy

If we prefer to move away from this approach, we could add a deprecation cycle to move from `to_bytes` to `as_bytes_like`

Reply:
I don't think `FsspecStore` works with `BytesLike` since it didn't accept a `memoryview` object when I tried it. So we probably need to keep `to_bytes()`.

Reply:
Think this would be [a good issue to raise with `fsspec`]( https://github.com/fsspec/filesystem_spec/issues/new ) if we can reproduce with it directly

If not, a Zarr issue would be welcome

Ideally these copies should be avoidable in the `FsspecStore` case as well

Reply:
Opened https://github.com/fsspec/s3fs/issues/959

Reply:
> Some possible names:

I'd also suggest `as_buffer_like`, because in my head "bytes" in Python is a type that always owns its own memory, while "buffer" doesn't necessarily own its memory, as in the buffer protocol or `collections.abc.Buffer`.